File: Metric.m
function Sum = Metric( M, G, eta, gamma )
%Metric Creator
%   Given a matrix, the input parameters(M,G,eta,gamma)
    [N,R] = size(G);
    Sum = 0;
    for i = 1:N
        for j = 1:R
            for k = 1:R
                if G(i,j) ~= 0 && G(i,k) ~= 0 && j ~= k   
                   s = Power(M(j,1), M(j,2), M(k,1), M(k,2), eta);
                   x = zeros(1:R);
                   for a = 1:R
                       if j ~= a && k ~= a
                          x(a) = Power(M(j,1), M(j,2), M(a,1), M(1,2),eta);
                       end
                   end
                   Sum = Sum + log(1+ SINR(s, x, gamma));
                   Sum = Sum(1);
                end               
            end
        end
    end




end

File: c_sim/metric.cpp
#include "metric.h"
#include "util.h"
#include "robot.h"
#include <cmath>
#include <vector> 
using namespace std;

Metric::Metric(vector<Robot> M, vector<Robot> G, double eta, double gamma ){
	int N = (int)G.size();
	int R = (int)(G[0].size());

	double sum = 0; 
	for (int i = 0; i < N; i++){
		for (int j = 0; j< R; j++){
			for (int k = 0; k < R; k++){
				if (G[i][j] != 0 && G[i][k] != 0 && j != k  ){
					double s = Power(M[j] , M[k]  , eta);
					vector<double> x; 
					for (int l = 0; l < R; l++ ){
						x.push_back(0);
					}
					for (int m = 0; m < R; m++){
						if (j != m && k != m){
							x[m] = Power(M[j], M[m], eta );
						}
					}
					sum = sum + log(1 + SINR(s, x, gamma) );
				}
			}
		}
	}
};